hosting/views.py

```diff
@@ -934,41 +936,55 @@ def post(self, request, *args, **kwargs):
             password=owner.password
         )
         vm_data = VirtualMachineSerializer(manager.get_vm(vm.id)).data
-        terminated = manager.delete_vm(
-            vm.id
-        )
+
+        terminated = manager.delete_vm(vm.id)
 
         if not terminated:
-            messages.error(
-                request,
-                'Error terminating VM %s' % (opennebula_vm_id)
-            )
-            return HttpResponseRedirect(self.get_success_url())
-        context = {
-            'vm': vm_data,
-            'base_url': "{0}://{1}".format(self.request.scheme,
-                                           self.request.get_host()),
-            'page_header': _('Virtual Machine Cancellation')
-        }
-        email_data = {
-            'subject': context['page_header'],
-            'to': self.request.user.email,
-            'context': context,
-            'template_name': 'vm_canceled',
-            'template_path': 'hosting/emails/',
-            'from_address': settings.DCL_SUPPORT_FROM_ADDRESS,
-        }
-        email = BaseEmail(**email_data)
-        email.send()
+            response['status'] = False
+            response['text'] = ugettext(
+                'Error terminating VM') + opennebula_vm_id
+        else:
+            context = {
+                'vm': vm_data,
+                'base_url': "{0}://{1}".format(self.request.scheme,
+                                               self.request.get_host()),
+                'page_header': _('Virtual Machine Cancellation')
+            }
+            email_data = {
+                'subject': context['page_header'],
+                'to': self.request.user.email,
+                'context': context,
+                'template_name': 'vm_canceled',
+                'template_path': 'hosting/emails/',
+                'from_address': settings.DCL_SUPPORT_FROM_ADDRESS,
+            }
+            email = BaseEmail(**email_data)
+            email.send()
 
-        messages.error(
-            request,
-            _('VM %(VM_ID)s terminated successfully') % {
-                'VM_ID': opennebula_vm_id}
+            # messages.error(
+            #     request,
+            #     _('VM %(VM_ID)s terminated successfully') % {
+            #         'VM_ID': opennebula_vm_id}
+            # )
+            t = 0
+            while True:
```

Comment:
Not sure where we are incrementing `t`. With the current code, I think we are making `get_vm` calls in a loop (and sleep would never be called) which is not good at all.

Reply:
yes, t has to be incremented.
for the `get_vm` calls the sleep is in `else` block for `try/except/else`, so its being called after every successfull `try` to `get_vm`

Reply:
I'll try to restructure the block to make it more readable